src/Instrumentation/Laravel/src/Hooks/Illuminate/Queue/Worker.php

```diff
@@ -0,0 +1,141 @@
+                $attributes = $this->buildMessageAttributes($queue, $job->getRawBody(), $job->getQueue());
+
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $span = $this->instrumentation
+                    ->tracer()
+                    ->spanBuilder(vsprintf('%s %s', [
+                        $attributes[TraceAttributes::MESSAGING_DESTINATION_NAME],
+                        'process',
+                    ]))
+                    ->setSpanKind(SpanKind::KIND_CONSUMER)
+                    ->setParent($parent)
+                    ->setAttributes($attributes)
+                    ->startSpan();
+
+                Context::storage()->attach($span->storeInContext($parent));
+
+                return $params;
+            },
+            post: function (QueueWorker $worker, array $params, $returnValue, ?Throwable $exception) {
+                $scope = Context::storage()->scope();
+                if (!$scope) {
+                    return;
+                }
+
+                $span = Span::fromContext($scope->context());
+                $job = ($params[1] instanceof Job ? $params[1] : null);
+
+                $span->setAttributes([
+                    'messaging.message.deleted' => $job?->isDeleted(),
+                    'messaging.message.released' => $job?->isReleased(),
+                ]);
+
+                $this->endSpan($exception);
+            },
+        );
+    }
+
+    private function hookWorkerGetNextJob(): bool
+    {
+        return hook(
+            QueueWorker::class,
+            'getNextJob',
+            pre: function (QueueWorker $worker, array $params, string $class, string $function, ?string $filename, ?int $lineno) {
+                /** @var Queue $connection */
+                $connection = $params[0];
+                $queue = $params[1];
+
+                $attributes = $this->buildMessageAttributes($connection, '', $queue);
+
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $span = $this->instrumentation
+                    ->tracer()
+                    ->spanBuilder(vsprintf('%s %s', [
+                        $attributes[TraceAttributes::MESSAGING_DESTINATION_NAME],
+                        TraceAttributeValues::MESSAGING_OPERATION_RECEIVE,
+                    ]))
+                    ->setSpanKind(SpanKind::KIND_CONSUMER)
+                    ->setAttributes($attributes)
+                    ->startSpan();
+
+                Context::storage()->attach($span->storeInContext(Context::getCurrent()));
+
+                return $params;
+            },
+            post: function (QueueWorker $worker, array $params, ?Job $job, ?Throwable $exception) {
+                $scope = Context::storage()->scope();
+                if (!$scope) {
+                    return;
+                }
+
+                // Discard empty receives.
+                if (!$job) {
+                    Context::storage()->destroy(
```

Comment:
not sure that destroy should be used here. I'm not 100% on how it works, but it looks like it will destroy a forked context (eg, for fiber-based processing). That's likely not doing anything here, but if there _were_ multiple forks, it could delete random other forks?
Maybe just detaching the scope will be enough. Thorough testing should uncover whether it works as expected, and DebugScope should complain if any mess was left behind.

Reply:
Thanks @brettmc, it looks like detach is all I needed.

I've added coverage of this via test, `test_it_drops_empty_receives`.